### PR TITLE
CSharp Module tweak & Unity Tutorial part 1, 2b, 3 tweaks

### PR DESCRIPTION
**Pull request by Lethalchip**
Small tweaks to CSharp module `index.md`:
- Added `.Tag` property to docs
- Revised example usage for both `Is*` usage and `.Tag` usage
- Fixed "Two" to "Four" in special reducer section
- Added examples for each `[SpacetimeDB.Reducer(ReducerKind.XXX)]`

Small tweak in Unity tutorial `part-1.md`:
- Fixed hyperlinks to Rust CSharp part-2

Fixed typos and mistakes throughout `part-2b-c-sharp.md`:
- Changed `Identity` to `uint` in Config table
- Updated `Throw` typo to `throw`
- Updated logging to correctly use `$""`
- Fixed `[SpacetimeDB.Reducer(ReducerKind.Connect)]` typo (it said Init before)
- Removed extraneous publish block, as there was already one at the end
- Fixed `UpdatePlayerLoginState()` error by switching `private` to `public` for compile

Fixed typos and clarified filenames in first-half of `part-3.md`:
- Renamed `GameManager.cs` throughout to `BitcraftMiniGameManager.cs`
- Added path clarifications to various filenames throughout
- Fixed typos on `PlayerComponent.FilterByOwnerId` => `PlayerComponent.FilterByIdentity` instead
- Removed `.FirstOrDefault();`'s from `UIUsernameChooser.cs` as not needed
- Corrected `.Text` to `.text` on `UIUsernameChooser.cs`

My changes stop at around the `**Append to bottom of RemotePlayer class in RemotePlayer.cs:**` - L230. 

The tutorial, or the [code files](<https://github.com/clockworklabs/com.clockworklabs.spacetimedbsdk/releases/latest>) provided are completely outdated and cannot be followed. I noticed that @dylanh724 has a PR #37 which likely fixes these issues, and probably some of the issues I fixed above. 

Anyway, I figured that I would provide my tweaks in case they are helpful to you, no worries either way.

### CODE REVIEW

File: docs/modules/c-sharp/index.md [hunk 1]
```diff
@@ -154,18 +154,34 @@ SpacetimeDB has support for tagged enums which can be found in languages like Ru
 
 To bridge the gap, a special marker interface `SpacetimeDB.TaggedEnum` can be used on any `SpacetimeDB.Type`-marked `struct` or `class` to mark it as a SpacetimeDB tagged enum. It accepts a tuple of 2 or more named items and will generate methods to check which variant is currently active, as well as accessors for each variant.
 
-It is expected that you will use the `Is*` methods to check which variant is active before accessing the corresponding field, as the accessor will throw an exception on a state mismatch.
+It is expected that you will use the `Is*` methods, or `.Tag` property to check which variant is active before accessing the corresponding field, as the accessor will throw an exception on a state mismatch.
```
Comment (RReverser) [inline docs/modules/c-sharp/index.md:157]: Oh no, `.Tag` was meant to be internal... either way, we have now made a larger breaking change to the tagged enum API, so this section will need to be reworked.

File: docs/unity/part-3.md [hunk 2]
```diff
@@ -199,15 +201,15 @@ public class RemotePlayer : MonoBehaviour
         canvas.worldCamera = Camera.main;
 
         // Get the username from the PlayerComponent for this object and set it in the UI
-        PlayerComponent? playerComp = PlayerComponent.FilterByEntityId(EntityId).FirstOrDefault();
+        PlayerComponent? playerComp = PlayerComponent.FilterByEntityId(EntityId);
```
Comment (RReverser) [inline docs/unity/part-3.md:204]: In 0.10 this will have to be `FindBy*`, for parity with C# module API.

File: docs/unity/part-3.md [hunk 3]
```diff
@@ -199,15 +201,15 @@ public class RemotePlayer : MonoBehaviour
...
         if (playerComp is null)
         {
-            string inputUsername = UsernameElement.Text;
+            string inputUsername = UsernameElement.text;
             Debug.Log($"PlayerComponent not found - Creating a new player ({inputUsername})");
             Reducer.CreatePlayer(inputUsername);
             
             // Try again, optimistically assuming success for simplicity
-            PlayerComponent? playerComp = PlayerComponent.FilterByEntityId(EntityId).FirstOrDefault();
+            playerComp = PlayerComponent.FilterByEntityId(EntityId);
```
Comment (RReverser) [inline docs/unity/part-3.md:212]: Same re:FindBy.

**Comment by RReverser** [comment]
Other than the outdated sections due to the new breaking API changes, this looks good, thanks!

File: docs/unity/part-1.md [hunk 4]
```diff
@@ -119,5 +119,5 @@ We chose ECS for this example project because it promotes scalability, modularit
 
 From here, the tutorial continues with your favorite server module language of choice:
 
-- [Rust](part-2a-rust)
-- [C#](part-2b-c-sharp)
+   - [Rust](/docs/unity/part-2a-rust.md)
+   - [C#](/docs/unity/part-2b-csharp.md)
```
Comment (RReverser) [inline docs/unity/part-1.md:123]: I think this change is wrong, the previous links work but the new ones don't.

File: docs/unity/part-2b-c-sharp.md [hunk 5]
```diff
@@ -334,12 +325,13 @@ public static void SendChatMessage(ReducerContext ctx, string text)
 ```
 
 ## Wrapping Up
-
+### Publishing a Module to SpacetimeDB
```
Comment (RReverser) [inline docs/unity/part-2b-c-sharp.md:328]: Please add an empty line before & after the heading (I don't remember if it's strictly required by Markdown, but at least it's a convention).

**Review by RReverser — APPROVED** [review]

**Review by jdetter — APPROVED** [review]
I have not tested this myself but the changes look reasonable to me. Thanks Chippy!

**Pull request merged**